scripts/install-multi-user.sh

```diff
@@ -199,6 +202,87 @@ ui_confirm() {
     return 1
 }
 
+printf -v _UNCHANGED_GRP_FMT "%b" $'\033[2m%='"$ESC" # "dim"
+# bold+invert+red and bold+invert+green just for the +/- below
+# red/green foreground for rest of the line
+printf -v _OLD_LINE_FMT "%b" $'\033[1;7;31m-'"$ESC ${RED}%L${ESC}"
+printf -v _NEW_LINE_FMT "%b" $'\033[1;7;32m+'"$ESC ${GREEN}%L${ESC}"
+
+_diff() {
+    # colorized diff comatible w/ pre `--color` versions
+    diff --unchanged-group-format="$_UNCHANGED_GRP_FMT" --old-line-format="$_OLD_LINE_FMT" --new-line-format="$_NEW_LINE_FMT" --unchanged-line-format="  %L" "$@"
+}
+
+confirm_rm() {
+    if ui_confirm "Can I remove $1?"; then
+        _sudo "to remove $1" rm "$1"
+    fi
+}
+
+confirm_edit() {
+    cat <<EOF
+
+It looks like Nix isn't the only thing here, but I think I know how to edit it
+out. Here's the diff:
+EOF
+
+    # could technically test the diff, but caller should do it
+    _diff "$1" "$2"
+    if ui_confirm "Does the change above look right?"; then
+        _sudo "remove nix from $1" cp "$2" "$1"
+    fi
+}
+
+_SERIOUS_BUSINESS="${RED}%s:${ESC} "
+password_confirm() {
+    if ui_confirm "Can I $1?"; then
+        # shellcheck disable=SC2059
+        sudo -kv --prompt="$(printf "${_SERIOUS_BUSINESS}" "Enter your password to $1")"
+    fi
+}
+
+# Support accumulating reminders over the course of a run and showing
+# them at the end. An example where this helps: the installer changes
+# something, but it won't work without a reboot. If you tell the user
+# when you do it, they may miss it in the stream. The value of the
+# setting isn't enough to decide whether to message because you only
+# need to message if you *changed* it.
+
+# reminders stored in array delimited by empty entry; if ! headless,
+# user is asked to confirm after each delimiter.
+_reminders=()
+((_remind_num=1))
+
+remind() {
+    if ! [ "${#_reminders}" = 0 ]; then
```

Comment:
```suggestion
    if [ "${#_reminders}" = 0 ]; then
      return
    fi
```
This could remove some intendantions.

Reply:
Not quite adopting this, but the discussion on it did make me realize this would be clearer as an arithmetic expression on `_remind_num`